lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -45,10 +51,34 @@ def on_block(node)
         private
 
         def unused_let_bang(node)
-          let_bang(node) do |method_send, method_name|
+          child_let_bang(node) do |method_send, method_name|
             yield(method_send) unless method_called?(node, method_name)
           end
         end
+
+        def child_let_bang(node)
+          enum = child_block_until_group(node).map(&method(:let_bang))
+
+          enum.compact.each do |node_with_name|
```

Comment:
Well, I told I'm missing a feature to make such things beautiful. Actually I think the code is now harder to read

Reply:
Indeed. Agree with you. Reverted these changes.